File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        StringBuffer sb = new StringBuffer(s);
        sb.reverse();
        s = sb.toString();
        String [] arr = s.split("\\s+");
        int n = arr.length;
        String res="";
        for(int i=0; i<n; i++)
        {
            StringBuffer ss = new StringBuffer(arr[i]);
            ss.reverse();
            if(i!=n-1)
                res += ss.toString() + " ";
            else
                res += ss.toString();
        }
        return res;
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string res = "";
        string word = "";
        int n = s.size();
        reverse(s.begin(), s.end());
        for(int i = 0; i < n; i++)
        {
            if(s[i] == ' ')
            {
                if(word.size() > 0)
                {
                    reverse(word.begin(), word.end());
                    res += word;
                    res += ' ';
                    word = "";
                }
            }
            else
            {
                word += s[i];
            }
        }
        reverse(word.begin(), word.end());
        res += word;
        n = res.size();
        if(res[n-1] == ' ') res.pop_back();
        return res;
    }
};